Compiler crash when overriding Java var-arg constructor

## Compiler version

3.2.2, 3.3.0-RC1

## Minimized code

Java:

```java
public class Foo {
  public Foo(String... xs) { }
}
```

Scala:

```scala
class Bar extends Foo {
  def this(xs: String*) = this()
}
```

## Output (click arrow to expand)
<details>

```scala
[info] compiling 1 Scala source and 1 Java source to E:\prog\jvm\scala\bugs\java-varargs\target\scala-3.2.2\classes ...
java.lang.AssertionError: assertion failed: constructor Bar in class Bar while running MegaPhase{crossVersionChecks, protectedAccessors, extmethods, uncacheGivenAliases, elimByName, hoistSuperArgs, forwardDepChecks, specializeApplyMethods, tryCatchPatterns, patternMatcher} on E:\prog\jvm\scala\bugs\java-varargs\src\main\scala\Bar.scala
[info] exception occurred while compiling E:\prog\jvm\scala\bugs\java-varargs\src\main\scala\Bar.scala
java.lang.AssertionError: assertion failed: constructor Bar in class Bar while compiling E:\prog\jvm\scala\bugs\java-varargs\src\main\java\Foo.java, E:\prog\jvm\scala\bugs\java-varargs\src\main\scala\Bar.scala
[error] ## Exception when compiling 2 sources to E:\prog\jvm\scala\bugs\java-varargs\target\scala-3.2.2\classes
[error] java.lang.AssertionError: assertion failed: constructor Bar in class Bar
[error] scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error] dotty.tools.dotc.transform.ForwardDepChecks.checkSelfConstructorCall(ForwardDepChecks.scala:117)
[error] dotty.tools.dotc.transform.ForwardDepChecks.transformApply(ForwardDepChecks.scala:123)
[error] dotty.tools.dotc.transform.ForwardDepChecks.transformApply(ForwardDepChecks.scala:120)
[error] dotty.tools.dotc.transform.MegaPhase.goApply(MegaPhase.scala:650)
```
</details>

--- a/compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala
+++ b/compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala
@@ -87,7 +87,8 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
    *  signatures of a Java varargs method and a Scala varargs override are not the same.
    */
   private def overridesJava(sym: Symbol)(using Context) =
-    sym.owner.info.baseClasses.drop(1).exists { bc =>
+    sym.memberCanMatchInheritedSymbols
+    && sym.owner.info.baseClasses.drop(1).exists { bc =>
       bc.is(JavaDefined) && {
         val other = bc.info.nonPrivateDecl(sym.name)
         other.hasAltWith { alt =>

--- a/tests/pos/i17066/Bar.scala
+++ b/tests/pos/i17066/Bar.scala
@@ -1,0 +1,2 @@
+class Bar extends Foo:
+  def this(xs: String*) = this()

--- a/tests/pos/i17066/Foo.java
+++ b/tests/pos/i17066/Foo.java
@@ -1,0 +1,3 @@
+public class Foo {
+  public Foo(String... xs) { }
+}

